src/acl/SquidErrorData.cc

```diff
@@ -52,11 +53,8 @@ ACLSquidErrorData::parse()
 
         if (err < ERR_MAX)
             errors.push_back(err);
-        else {
-            debugs(28, DBG_CRITICAL, "FATAL: Invalid squid error name");
-            if (!opt_parse_cfg_only)
-                self_destruct();
-        }
+        else
+            throw Cfg::FatalError(ToSBuf("invalid Squid error name ", token));
```

Comment:
This is one of those ad-hoc situations where -k parse continues after finding broken config to show the user more mistakes. It just happens to have been using self_destruct locally instead of a fatal*() call, or skipping some other error handling path.

Reply:
If we want to make this particular input (wrong/unknown/unsupported ERR_* name in squid.conf ACL list) just a WARNING situation;  then IMO there should not be any exception thrown at all. Just a level-2 debugs message stating the issue and a hint for fixing.

Reply:
> This is one of those ad-hoc situations where -k parse continues after finding broken config to show the user more mistakes.

Yes, of course.

> If we want to make this particular input just a WARNING situation ...

IMO, we do not. Lack of the right error page is often a critical (and otherwise hard to detect) error. Please do not relax this existing check.

Going forward, I see two acceptable solutions here:

1. Always throw a fatal error. This approach will kill the existing (useful) functionality, but if you insist on not doing [step 2](https://github.com/squid-cache/squid/pull/574#discussion_r397964286) in this PR, then I can live with that (in this context). This is what PR already does. If you insist on this and do not move to step 2 anywhere else in the PR, I will consider this change request addressed. Just add a TODO comment in this case so that we can find  it when we go to step 2. Something along the "make less fatal with opt_parse_cfg_only" lines. An opt_parse_cfg_only mention would be the key for us to look for.

2. Always throw a non-fatal error. This is the solution I would prefer because it keeps the existing (useful) functionality -- allowing things like `-k parse` to continue upon catching this particular error. See [step 2](https://github.com/squid-cache/squid/pull/574#discussion_r397964286) in another change request.

Reply:
Given that this code already implements your "acceptable option" 2 - I do not see any point in continuing this discussion.

Reply:
```C++
throw Cfg::FatalError(...);
```

>> Alex: 2. Always throw a non-fatal error.

> Amos: Given that this code already implements your "acceptable option" 2 - I do not see any point in continuing this discussion.

The code throws a fatal error (i.e. an object of the type called FatalError). My option 2 is about throwing a non-fatal error. Thus, this code does _not_ implement option 2. Am I missing something in this seemingly basic logic sequence? Its been two years so I may have forgotten some hidden dependencies.

Please note that Cfg::FatalError() _implementation_ is irrelevant to this change request. Its problems are detailed in other change requests IIRC.